File: App/src/app/chat/entities/message.entity.ts
import { BaseEntity } from '../../shared/entities/base.entity';
import { UserEntity } from '../../user/entities/user.entity';
import { ChatEntity } from './chat.entity';

export interface MessageEntity extends BaseEntity {
  content: string;
  timestamp: Date;
  userId: number;
  user?: UserEntity;
  chatId: number;
  chat?: ChatEntity;
}

File: Api/src/Api.Domain/Entities/MessageEntity.cs
using System;
using System.ComponentModel.DataAnnotations;
using Newtonsoft.Json;

namespace Domain.Entities {
    public class MessageEntity : BaseEntity
    {
        [Required]
        [MaxLength(2000)]
        public string Content { get; set; }
        
        public DateTime Timestamp { get; set; }

        [Required]
        public int UserId { get; set; }

        public UserEntity User { get; set; }

        [Required]
        public int ChatId { get; set; }

        [JsonIgnore]
        public ChatEntity Chat { get; set; }
    }
}
